tests/integration/suite/actors/reminders/scheduler/upsert.go

```diff
@@ -0,0 +1,128 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	rtv1 "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/framework/process/http/app"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/process/scheduler"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(upsert))
+}
+
+type upsert struct {
+	place     *placement.Placement
+	scheduler *scheduler.Scheduler
+
+	daprd        *daprd.Daprd
+	methodcalled atomic.Uint32
+}
+
+func (u *upsert) Setup(t *testing.T) []framework.Option {
+	configFile := filepath.Join(t.TempDir(), "config.yaml")
+	require.NoError(t, os.WriteFile(configFile, []byte(`
+apiVersion: dapr.io/v1alpha1
+kind: Configuration
+metadata:
+  name: schedulerreminders
+spec:
+  features:
+  - name: SchedulerReminders
+    enabled: true`), 0o600))
+
+	app := app.New(t,
+		app.WithConfig(`{"entities": ["myactortype"]}`),
+		app.WithHandlerFunc("/actors/myactortype/myactorid/method/remind/xyz",
+			func(_ http.ResponseWriter, r *http.Request) { u.methodcalled.Add(1) }),
+		app.WithHandlerFunc("/actors/myactortype/myactorid/method/foo", func(http.ResponseWriter, *http.Request) {}),
+	)
+
+	u.scheduler = scheduler.New(t)
+
+	u.place = placement.New(t)
+	u.daprd = daprd.New(t,
+		daprd.WithConfigs(configFile),
+		daprd.WithInMemoryActorStateStore("mystore"),
+		daprd.WithPlacementAddresses(u.place.Address()),
+		daprd.WithSchedulerAddresses(u.scheduler.Address()),
+		daprd.WithAppPort(app.Port()),
+	)
+
+	return []framework.Option{
+		framework.WithProcesses(u.scheduler, u.place, app, u.daprd),
+	}
+}
+
+func (u *upsert) Run(t *testing.T, ctx context.Context) {
+	u.scheduler.WaitUntilRunning(t, ctx)
+	u.place.WaitUntilRunning(t, ctx)
+	u.daprd.WaitUntilRunning(t, ctx)
+
+	gclient := u.daprd.GRPCClient(t, ctx)
+	_, err := gclient.RegisterActorReminder(ctx, &rtv1.RegisterActorReminderRequest{
+		ActorType: "myactortype",
+		ActorId:   "myactorid",
+		Name:      "xyz",
+		DueTime:   "1h",
+	})
+	require.NoError(t, err)
+
+	time.Sleep(time.Second * 2)
+	require.Equal(t, 0, int(u.methodcalled.Load()))
+
+	_, err = gclient.RegisterActorReminder(ctx, &rtv1.RegisterActorReminderRequest{
+		ActorType: "myactortype",
+		ActorId:   "myactorid",
+		Name:      "xyz",
+		DueTime:   "1s",
+	})
+	require.NoError(t, err)
+
+	assert.Eventually(t, func() bool {
```

Comment:
I think this check has the potential to be flaky under some conditions, but given the reminder has a whole second until it triggers, it should be ok